lib/dataSource.js

```diff
@@ -16,138 +16,123 @@ var fluid = require("infusion"),
- * entries in "termMap" which themselves take the form %path will be looked up within the argument `directModel`. All such entries will then be URI Encoded, unless their value
- * the termMap is prefixed with `noencode:`. Secondly,
- * this resolved termMap will be used for a round of the standard fluid.stringTemplate algorithm applied to the supplied URL. Finally, any argument `expand` will be
- * used to expand the entire URL.
- * @param  {Function} expand an optional function which applies further expansion to the URL before returning (currently urlExpander's expand method - this will be removed)
- * @param  {String} url a url or path to expand.
- * @param  {JSON} termMap a set of configuration for urlExpander's expand method. Any values which begin with the prefix `noencode:` will have this prefix stripped off, and
- * URI encoding will not be applied to the substituted value.
- * @param  {Object} directModel a set of data used to expand the url template.
- * @return {String} resolved/expanded url.
- */
-kettle.dataSource.urlResolver.resolve = function (expand, url, termMap, directModel) {
-    var map = fluid.transform(termMap, function resolve(entry) {
-        entry = "" + entry;
-        var encode = true;
-        if (entry.indexOf("noencode:") === 0) {
-            encode = false;
-            entry = entry.substring("noencode:".length);
-        }
-        if (entry.charAt(0) === "%") {
-            entry = fluid.get(directModel, entry.substring(1));
-        }
-        if (encode) {
-            entry = encodeURIComponent(entry);
-        }
-        return entry;
-    });
-    var replaced = fluid.stringTemplate(url, map);
-    replaced = (expand || fluid.identity)(replaced);
-    return replaced;
-};
+fluid.defaults("kettle.dataSource.encoding.none", {
+    gradeNames: "fluid.component",
+    invokers: {
+        parse: "fluid.identity",
+        render: "fluid.identity"
+    },
+    contentType: "text/plain"
+});
 
 
-/** STANDARD DATASOURCE GRADE HIERARCHY **/
+/** Definitions for parsing JSON using jsonlint to render errors **/
 
+kettle.dataSource.JSONParseErrors = [];
 
-// This table of standard data converter priorities will be removed when we have
-// FLUID-5506 completed
-// These are the priorities applied on READ from the dataSource
-kettle.dataSource.priorities = {
-    JSON: 100, // high priority - do this conversion first
-    CouchDB: 90
+kettle.dataSource.accumulateJSONError = function (str, hash) {
+    var error = "JSON parse error at line "+ hash.loc.first_line + ", col " + hash.loc.last_column + ", found: \'"+ hash.token + "\' - expected: "+ hash.expected.join(", ");
+    kettle.dataSource.JSONParseErrors.push(error);
 };
 
-kettle.dataSource.parse = function (string) {
-    var togo;
-    if (!string) {
-        return undefined;
-    }
+// Adapt to shitty integration model of JISON-based parsers - beware that any other user of this module will find it permanently corrupted
```

Comment:
I left this comment elsewhere, but as it was on an outdated diff, I'll revisit here.  If you are seeing shortcomings with JSONlint, it could be worth considering [AJV](https://github.com/epoberezkin/ajv).  It can be instructed to continue past the first error, and is also evolving closely in parallel with the standard itself.